library/src/test/java/io/ona/kujaku/layers/ArrowLineLayerTest.java

```diff
@@ -313,8 +312,143 @@ public void setLayerPropertiesFromBuilder() {
         assertEquals(propertyValues.get("line-width").value, lineWidth);
     }
 
+    @Test
+    public void constructorShouldThrowExceptionWhenDateTimeFormatIsNotSet() {
```

Comment:
You can use  `@Test(expected = InvalidArrowLineConfig.class)` to simplify testing an exception is thrown as expected.

Reply:
It doesn't work with the current dependencies which end up using a method that is not available  or supported for the added dependencies.

Reply:
I find that if `InvalidArrowLineConfigException` were to extend `RuntimeException` or `IllegalArgumentException`, then you can use that syntax.

Do either of those work?

Reply:
https://stackoverflow.com/questions/2517965/what-exception-to-throw-when-an-important-parameter-dependency-is-missing

Reply:
If you were to run the tests using that annoation, they will not run because the dependencies tend to mix up the versions which is currently not fixable. That is what I mean. You can use the annotation and run the tests yourself to see

Reply:
I had tried it out already and it works.

Please test and see, extend `IllegalArgumentException` in `InvalidArrowLineConfigException` and add `@Test(expected = InvalidArrowLineConfigException.class)`.

Reply:
Yea, that works. Sorry, I meant the `ExpectedException` rule so that we can assert the message also.

Changing to `IllegalArgumentException` or any other child of `RuntimeException` is not ideal  for this situation since the app can still work OK without the `ArrowLineLayer` only that the layer will not be drawn. The error is also avoidable.

Reply:
OK.